app/utils/press-release-snapshot.js

```diff
@@ -1,39 +1,68 @@
 import { tracked } from '@glimmer/tracking';
 
+function serializePublicationChannels(publicationChannels) {
+  let publicationChannelsStringArray = [];
+  let publicationChannelsString = '';
+
+  publicationChannelsStringArray = publicationChannels.map(channel => channel.uri);
+
+  publicationChannelsString = publicationChannelsStringArray
+                                .sort()
+                                .join('+');
+
+  return publicationChannelsString;
+}
+
 /**
  * Snapshot of a PressRelease (Pers bericht) record and related records to keep track of changes,
  * because ember-data lacks dirty tracking for relationships and attributes of type 'array'.
  *
+ * Tracks relationship publicationChannels, to be able to find any changes, and detect those in the isDirty check.
+ * 
  * Contains application-specific logic to track the dirty state of relationships.
  * If the model of a PressRelease changes in the future, this class will probably require an update as well.
 */
 export default class PressReleaseSnapshot {
   @tracked pressRelease;
+  @tracked publicationChannels = [];
 
   constructor(pressRelease) {
     this.pressRelease = pressRelease;
   }
 
+
   async commit() {
-    // commit changes when pressRelease gets relations => e.g. source-snapshop.js
+    this.publicationChannels = await this.pressRelease.publicationChannels.map(channel => channel);
```

Comment:
`myArray.slice(0)` is een meer gebruikelijke manier om een kopie te maken van een array in javascript

Reply:
thanks voor de tip ;)